conjure/serverwriter.go

```diff
@@ -250,20 +250,39 @@ func astForHandlerMethodDecodeBody(methodBody *jen.Group, argDef *types.Endpoint
 	if argDef == nil {
 		return
 	}
+
+	varName := transforms.SafeName(argDef.Name)
+
+	emptyBodyCondition := jen.Id(reqName).Dot("Body").Op("!=").Nil().Op("&&").
```

Comment:
Just so I am clear on this piece, if a user provides a nil input argument, it immediately gets mapped to `http.NoBody` and thus we need the `&& req.Body != http.NoBody` so that the argument passed into the impl below is correctly set to nil rather than the `http.NoBody` struct?

Reply:
Yep, exactly. I think the reason the stdlib uses NoBody instead of nil is to avoid panics reading from req.Body. I don't think our implementations need to be exposed to that detail, so we use nil.